WordPress/src/main/java/org/wordpress/android/ui/jetpack/JetpackCapabilitiesUseCase.kt

```diff
@@ -31,58 +30,62 @@ class JetpackCapabilitiesUseCase @Inject constructor(
     @Suppress("unused") private val siteStore: SiteStore,
     private val dispatcher: Dispatcher,
     private val appPrefsWrapper: AppPrefsWrapper,
-    private val currentDateProvider: CurrentTimeProvider,
-    @param:Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+    private val currentDateProvider: CurrentTimeProvider
 ) {
-    private var continuation: Continuation<OnJetpackCapabilitiesFetched>? = null
+    private var continuation: HashMap<Long, Continuation<OnJetpackCapabilitiesFetched>?> = hashMapOf()
 
-    suspend fun getJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts {
-        val capabilities = withContext(bgDispatcher) {
-            getOrFetchJetpackCapabilities(remoteSiteId)
+    suspend fun getJetpackPurchasedProducts(remoteSiteId: Long) = flow {
+        emit(getCachedJetpackPurchasedProducts(remoteSiteId))
+        if (!hasValidCache(remoteSiteId)) {
+            emit(fetchJetpackPurchasedProducts(remoteSiteId))
         }
-        return JetpackPurchasedProducts(
-                scan = capabilities.find { SCAN_CAPABILITIES.contains(it) } != null,
-                backup = capabilities.find { BACKUP_CAPABILITIES.contains(it) } != null
-        )
     }
 
-    suspend fun getOrFetchJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return if (hasValidCache(remoteSiteId)) {
-            getCachedJetpackCapabilities(remoteSiteId)
-        } else {
-            fetchJetpackCapabilities(remoteSiteId)
-        }
-    }
+    fun getCachedJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts =
```

Comment:
getCachedJetpackPurchasedProducts and fetchJetpackPurchasedProducts is used on the legacy (currently used) My Site fragment. It doesn't use flow -> it needs to synchronously load cached value since the state is refreshed from onResume().